content/Secure Coding/Buffer Overflow/CS0 C++/checklist.js

```diff
@@ -16,115 +16,147 @@ $(document).ready(function() {
+					$('.rangepop').hide();
+					$(".span-"+name).click(); //click the empty span so that next question is asked.
+					$('.popover').css('background-color', 'white');
+				} else {
+					$('.popover').css('background-color', 'lightpink');
 				}
+			});
+		} else { //NORMAL QUESTION
 
-					return next;
+			waitingOn = [];
+			$(".span-"+name).each(function(index) {
+				if( $(this).hasClass(next+"-"+name) ) {
+					waitingOn.push(index);
 				}
+			});
+		}	
+		//removes highlighted sections so user is not seeing correct answers of previous question
+		$(".span-" + name).removeClass("si-code-clicked");
+		return next;
+	}
 
+	current = advance();
 
-				current = advance();
-
-
+	//Handles score for each question
 	$(".span-"+name).each(function(index) {
 		spans.push($(this));
-		//console.log(spans);
+		
 		$(this).on('click', function() {
 			span = $(this);
 
 			// continue if user is currently supposed to click this span, and hasn't already
-			if( span.hasClass(current+"-"+name) && $.inArray(index,clicked) < 0 ) {
-				clicked.push(index);					// 		note that it's been clicked, programmatically
-				span.addClass("si-code-clicked");		//		note that it's been clicked, graphically
-				if(span.hasClass(name+"-var-modify-"+name)|| span.hasClass(name+"-var-array-"+name)) { //needs to be changed for vulnerabilities
-					span.addClass("si-code-vulnerability"); //	some spans get extra graphics to indicate vulnerability
-				}
-
+			if(span.hasClass(current+"-"+name) && $.inArray(index,clicked) < 0 ) {
+				clicked.push(index);  // note that it's been clicked, programmatically
+				span.addClass("si-code-clicked");  // note that it's been clicked, graphically
 
 				// Check if 'current' question is finished yet
 				var finished = true;
+
+				// logic for increasing the progress bar
+				let currentProgress = ((clicked.length - previousWaitingOnCount)/waitingOn.length)*100;
+				document.getElementById(current + "-progress-data-label").style.width = `${currentProgress}%`;
+
 				for(i in waitingOn) {
 					finished &= $.inArray(waitingOn[i],clicked) >= 0;
 				}
 
 				// if it is, go to next question
-				if(finished) current = advance(current);
+				if (finished) {
+					// highlights correct answer before moving on to next question
+					setTimeout(function(){
+						previousWaitingOnCount += waitingOn.length;
+						current = advance(current);
+					}, 500);
+				}
 			}
 		});
 	});
 
-	function showPopOver(span) { //showsPopover over span
+	//showsPopover over span
+	function showPopOver(span) { 
 	    var offset = span.offset();
 	    var position = span.position();
 	    var theHeight = $('.popover').height();
 	    $('.rangepop').show();
 			$(".popover").css({
             'position': 'absolute',
             'top': (position.top-(theHeight/2)-5) + 'px',
-            'left': (position.left+50) + 'px'
-      });
+            'left': (position.left+100) + 'px'
+    	});
```

Comment:
If you see a popover covering the code, adjust the pixels here to move it more to the left